### PR TITLE
Update documentation for requisites

### PR DESCRIPTION
**Pull request by MTecknology**
### What does this PR do?

Organize and clean-up documentation.

A lot of the requisites didn't make sense to me and I struggled to understand why one would be chosen over the other because they mostly sounded the same. I tried to work through a layman explanation of what makes each requisite special.

I also attempted to consolidate repetitive explanations, remove overly-technical bits, and re-arrange things so they're easier to navigate.

Hopefully all of my changes are helpful and technically accurate, but added explanations are to the best of my comprehension, so review would be excellent.

### Commits signed with GPG?

Yes

### CODE REVIEW

**Comment by MTecknology** [comment]
@garethgreenaway Could you please make sure the description of ``req_any`` is accurate?

File: doc/ref/states/requisites.rst [hunk 1]
```diff
@@ -7,80 +7,58 @@ Requisites and Other Global State Arguments
...
-
-Requisites come in two types: Direct requisites (such as ``require``),
-and requisite_ins (such as ``require_in``). The relationships are
-directional: a direct requisite requires something from another state.
-However, a requisite_in inserts a requisite into the targeted state pointing to
-the targeting state. The following example demonstrates a direct requisite:
+The Salt requisite system is used to create relationships between states. This
+provides a method to easily define inter-dependencies between states. These
+dependencies are expressed by declaring the relationships using state names
+and ID's or names. The generalized form of a requisite target is ``<state name>:
```
Comment (cachedout) [inline doc/ref/states/requisites.rst]: `IDs`

File: doc/ref/states/requisites.rst [hunk 2]
```diff
@@ -143,100 +121,69 @@ so either of the following versions for "Extract server package" works:
...
-* ``onfail_in``
-
-There are several corresponding requisite_any statements:
+* `require <requisites-require>`: Require success from another state
+* `onchanges <requisites-onchanges>`: Execute if target has changes
+* `watch <requisites-watch>`: Similar to require, but invokes ``mod_watch`` behavior
+* `listen <requisites-listen>`: Similar to require, but invokes ``mod_wait`` behavior
+* `prereq <requisites-prereq>`: Similar to onchanges with reversed execution order
+* `onfail <requisites-onfail>`: Execute if target state fails
+* `use <requisites-use>`: Copy arguments from another state
```
Comment (cachedout) [inline doc/ref/states/requisites.rst]: I think this could be more descriptive.

File: doc/ref/states/requisites.rst [hunk 3]
```diff
@@ -248,62 +195,117 @@ including the sls file and then setting a state to ``require`` the included sls
...
 .. _requisites-watch:
 
 watch
 ~~~~~
 
-``watch`` statements are used to add additional behavior when there are changes
-in other states.
+A ``watch`` requisite is used to add additional behavior when there are changes
+in other states. This is done using the ``mod_watch`` function available from
+the execution module and will execute any time a ``watch``\ed state changes.
```
Comment (cachedout) [inline doc/ref/states/requisites.rst]: Is this escape char intentional?

Comment (MTecknology) [reply doc/ref/states/requisites.rst]: In my viewer, it was required to render \`\`watch\`\`ed as ``watch``ed. Apparently that's a local limitation. I'll remove it.

File: doc/ref/states/requisites.rst (hunk 2 above)
Comment (MTecknology) [reply doc/ref/states/requisites.rst]: How's the second attempt?

**Review by garethgreenaway — APPROVED** [review]
The description of require_any looks accurate.

**Comment by MTecknology** [comment]
@garethgreenaway Excellent, thanks for checking!

**Review by cachedout — APPROVED** [review]

**Pull request merged**